scripts/init-mptcp.sh

```diff
@@ -0,0 +1,33 @@
+echo "Setting up MPTCP..."
+
+vagrant ssh client <<EOF
+    # Sometimes the ip addresses aren't assigned by virtualbox, so do it manually
+    sudo ip addr add 192.168.56.100/24 dev eth1
+    sudo ip addr add 192.168.57.100/24 dev eth2
+    sudo ip addr add 192.168.58.100/24 dev eth3
+
+    # Configure MPTCP
+    sudo sysctl net.mptcp.enabled=1
+    sudo ip mptcp limits set subflow 2
+    sudo ip mptcp limits set add_addr_accepted 2
+    sudo ip mptcp endpoint add 192.168.57.100 dev eth2 subflow fullmesh
```

Comment:
you might want to add an endpoint with eth2 included: to be also used as a "fullmesh". Or you remove 'fullmesh', not to create too many links. `fullmesh` is usually not needed.

Reply:
> you might want to add an endpoint with eth2 included
I didn't really understand, what are you meaning about this ?

Reply:
It is not clear to me why you need `fullmesh`. With `fullmesh`, it means that he MPTCP path manager will try to create an additional subflow for each known peer address, using this endpoint as the source IP address.

So here, when an ADD_ADDR will be received by the client, it will try to create a subflow with eth2 and eth3, not eth1. (depending on the limits).

I guess in your setup, you will have the server sending an ADD_ADDR, and the client will create two subflows from eth2 and eth3 (because of the fullmesh flag) to this single address. If it is what you want, that's OK, but a bit "strange".

Reply:
So if I understand correctly, by setting full mesh, it will explicitly create the flows client:eth1-server:eth2 and client:eth1-server:eth3, instead of client:eth2-server:eth2 and client:eth3-server:eth3 ?

Reply:
The goal of fullmesh is to create all possible paths. It means two things:
- all endpoints with the subflow and fullmesh flags will be use to create subflows to the initial address of the server
- Upon the reception of an ADD_ADDR, the PM will look at endpoints with the fullmesh flag to create new subflows (or use the routing rule to find the source address).

Here, I see that you are using fullmesh as a workaround for this specific lab setup. That's ok, you can keep them. That's not a real setup, that's why it strange ;)

Reply:
Thanks for the explanations, it is clearer now !